File: 74-search-a-2d-matrix/74-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        i,j=0,len(matrix[0])-1
        while i<len(matrix) and j>=0:
            if matrix[i][j]==target:
                return True
            elif matrix[i][j]>target:
                j-=1
            elif matrix[i][j]<target:
                i+=1
        return False
        

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m=matrix.length, n=matrix[0].length;
        int lo=0,hi=(m*n)-1;
        while(lo<=hi){
            int mid=lo+(hi-lo)/2;
            if(matrix[mid/n][mid%n]==target) return true;
            else if(matrix[mid/n][mid%n]<target) lo=mid+1;
            else if(matrix[mid/n][mid%n]>target) hi=mid-1;
        }
        return false;
    }
}